fix save virtual category conditions via api

androshchuk opened the pull request.

romainruaud commented:
KO for me.

I did not test the save via API but started to test if I was still able to create & save a virtual category via the back-office, and it appears to be broken : 

```
[2019-07-31 11:50:23] main.CRITICAL: Notice: Undefined index: condition in /var/www/magento/src/vendor/smile/elasticsuite/src/module-elasticsuite-virtual-category/Model/Category/Attribute/VirtualRule/SaveHandler.php on line 67 {"exception":"[object] (Exception(code: 0): Notice: Undefined index: condition in /var/www/magento/src/vendor/smile/elasticsuite/src/module-elasticsuite-virtual-category/Model/Category/Attribute/VirtualRule/SaveHandler.php on line 67 at /var/www/magento/src/vendor/magento/framework/App/ErrorHandler.php:61)"} []
```

androshchuk commented:
I fixed it and I added my postman collection for API testing  
[magento.postman_collection.json.zip](https://github.com/Smile-SA/elasticsuite/files/3452453/magento.postman_collection.json.zip)
